src/state.rs

```diff
@@ -22,6 +23,16 @@ pub struct Heap {
     heap: Vec<u8>,
 }
 
+impl Heap {
+    pub fn len(&self) -> usize {
+        self.heap.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.heap.is_empty()
+    }
+}
+
```

Comment:
There is already an impl Heap down this file, combine both

Reply:
Missed it!